### PR TITLE
Update documentation for development environment setup

### PR DESCRIPTION
**Pull request by jen-rich**
Added fixes for issues with setup on Windows

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/developer_guide/development-workflow.md [hunk 1]
```diff
@@ -106,6 +106,23 @@ You can build and install all Elyra packages with:
 ```bash
 make clean install
 ```
+##### Windows Users
+
+Windows does not recognize `make`
```
Comment (karlaspuldaro) [inline docs/source/developer_guide/development-workflow.md:111]:
```suggestion
Windows does not recognize `make` command
```

File: docs/source/developer_guide/development-workflow.md [hunk 2]
```diff
@@ -106,6 +106,23 @@ You can build and install all Elyra packages with:
...
+
+You can get Windows to recognize make with:
```
Comment (karlaspuldaro) [inline docs/source/developer_guide/development-workflow.md:113]: Maybe to avoid repetition, we could rephrase it to something like

```suggestion
You can install `make` on Windows with:
```

**Comment by lresende** [comment]
Was this closed by mistake?

**Review by karlaspuldaro — APPROVED** [review]
LGTM 👍  Thank you Jennifer!

File: docs/source/developer_guide/development-workflow.md [hunk 3]
```diff
@@ -106,6 +106,23 @@ You can build and install all Elyra packages with:
...
+If you have Chocolatey:
+
+```bash
+choco install make
+```
+
+or 
+
+```bash
+conda install m2-base
```
Comment (lresende) [inline docs/source/developer_guide/development-workflow.md:124]: 'm2-base' by itself does not contain 'make', you need to also install 'm2-make' to enable the make command in windows. I would suggest also adding '-y' on the command to avoid confirmation prompt.

**Comment by lresende** [comment]
After the ' conda install -y m2-base m2-make' I also had the following:
`SSLError("Can't connect to HTTPS URL because the SSL module is not available.")'`

which I was able to workaround by exporting the following env variable:
`set  CONDA_DLL_SEARCH_MODIFICATION_ENABLE=true`

And then `make install` still failed with:
```
error An unexpected error occurred: "ENOENT: no such file or directory, chmod 'D:\\opensource\\elyra\\elyra\\node_modules\\.bin\\jest'".
info If you think this is a bug, please open a bug report with the information provided in "D:\\opensource\\elyra\\elyra\\yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```


And the log

**Review by lresende — CHANGES_REQUESTED** [review]
We are discussing some issues I am seeing following these instructions. Might require additional details.

**Comment by lresende** [comment]
On a fresh windows install, I tried the following:
```
Install git from https://git-scm.com/download/win
Install miniconda from https://docs.conda.io/en/latest/miniconda.html
conda create -y --name dev python=3.7 jupyterlab pandas numpy r r-essentials r-irkernel
conda install m2-base m2-make 
conda install nodejs
conda install yarn

mkdir ~/.npm-global
npm config set scripts-prepend-node-path auto
prefix=c:\UsersAdministrator\.npm-global

Install OpenSSL from https://slproweb.com/products/Win32OpenSSL.html
```

And then, trying to ` make install`  I get the following:
```
export PATH=$(pwd)/node_modules/.bin:$PATH && lerna run build
lerna notice cli v3.22.1
lerna info Executing command in 8 packages: "yarn run build"
lerna ERR! yarn run build exited 1 in '@elyra/ui-components'
lerna ERR! yarn run build stdout:
yarn run v0.25.2
$ tsc
src/DropDown.tsx(26,8): error TS2307: Cannot find module '@material-ui/core' or its corresponding type declarations.
src/DropDown.tsx(27,30): error TS2307: Cannot find module '@material-ui/lab' or its corresponding type declarations.
src/DropDown.tsx(46,34): error TS7006: Parameter '_theme' implicitly has an 'any' type.
src/DropDown.tsx(122,27): error TS7006: Parameter 'params' implicitly has an 'any' type.
src/JSONComponent.tsx(18,22): error TS2307: Cannot find module 'react-json-tree' or its corresponding type declarations.
src/JSONComponent.tsx(64,25): error TS7006: Parameter 'type' implicitly has an 'any' type.
src/JSONComponent.tsx(64,31): error TS7006: Parameter 'data' implicitly has an 'any' type.
src/JSONComponent.tsx(64,37): error TS7006: Parameter 'itemType' implicitly has an 'any' type.
src/JSONComponent.tsx(64,47): error TS7006: Parameter 'itemString' implicitly has an 'any' type.
src/JSONComponent.tsx(77,26): error TS7031: Binding element 'label' implicitly has an 'any' type.
src/JSONComponent.tsx(77,33): error TS7031: Binding element 'type' implicitly has an 'any' type.
src/JSONComponent.tsx(80,25): error TS7006: Parameter 'raw' implicitly has an 'any' type.
src/TextInput.tsx(24,8): error TS2307: Cannot find module '@material-ui/core' or its corresponding type declarations.
src/TextInput.tsx(25,43): error TS2307: Cannot find module '@material-ui/icons' or its corresponding type declarations.
src/TextInput.tsx(42,34): error TS7006: Parameter '_theme' implicitly has an 'any' type.
src/TextInput.tsx(85,22): error TS7006: Parameter 'event' implicitly has an 'any' type.
src/TextInput.tsx(102,39): error TS7006: Parameter 'event' implicitly has an 'any' type.
src/ThemeProvider.tsx(23,8): error TS2307: Cannot find module '@material-ui/core' or its corresponding type declarations.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build stderr:
(node:4736) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
error Command failed with exit code 1.
``` 
 
Any suggestions?

**Comment by karlaspuldaro** [comment]
@jen-rich could you run a build with `make install` and post your log here?

File: docs/source/developer_guide/development-workflow.md (hunk 3 above)
Comment (bourdakos1) [reply docs/source/developer_guide/development-workflow.md:124]: Could this be a case for `npm scripts` over `make`? I guess this isn't a huge hurdle, but just one less dependency to install for windows users

**Comment by jen-rich** [comment]
Reinstalled Jupyter via conda and it seemed to fix issues with some missing packages (pysqlite and zeromq). Originally had installed it via Anaconda.

**Comment by akchinSTC** [comment]
@jen-rich - gentle nudge, any updates on this pull request?

**Comment by jen-rich** [comment]
> @jen-rich - gentle nudge, any updates on this pull request?

I am making progress. Updating Jupyter got rid of the error with pysqlite and a recent update to pip got rid of some more errors. I was also able to eliminate a zmq error by reinstalling zmq via `pip install zeromq`. Unfortunately, I recently encountered this error `module 'zmq' has no attribute 'REQ'`, so I am looking into it

**Comment by jdcai** [comment]
Hi, I just went through the development environment set up on Windows though it was not a fresh windows install. This is what I went through and got it building.

I went through the `Setting up your development environment` section on https://elyra.readthedocs.io/en/latest/developer_guide/development-workflow.html#setting-up-your-development-environment which worked fine. I already had yarn installed before hand and it didn't seem like I had to do anything else regarding yarn.

I then did `conda install m2-base m2-make` to get `make` working on Windows.

I ran `make clean install` and got this error
```
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/git-python/
```
I fixed this error by adding Miniconda3 to my PATH
C:\Miniconda3
C:\Miniconda3\Scripts
C:\Miniconda3\Library\bin  
https://stackoverflow.com/a/54897379

After that I ran `make clean install` again and got a win32 error which was fixed by running `conda install pywin32`
https://stackoverflow.com/a/64359808

Running `make clean install` worked properly and I could run `make watch` and `jupyter lab --watch`. Then I was able to see the UI in my browser.

**Comment by karlaspuldaro** [comment]
I also tried the steps @jdcai suggested but I'm still having the exact same error log @lresende posted in [this earlier comment](https://github.com/elyra-ai/elyra/pull/1431#issuecomment-803786245). 
It could be related to something still missing on a fresh new windows install on this build step in [Makefile](https://github.com/elyra-ai/elyra/blob/master/Makefile):
```
build-ui: yarn-install lint-ui # Build packages
	export PATH=$$(pwd)/node_modules/.bin:$$PATH && lerna run build
```
